File: GTP2.5/gtp1.cpp
#include<iostream>
#include<string.h>

using namespace std;

void input(string names[], int codes[]) {
        for(short i = 0; i < 5; i++) {
                cout << "Ingrese el nombre del empleado " << i+1 << ": ";
                cin >> names[i];

                cout << "Ingrese el código de ese empleado: ";
                cin >> codes[i];
        }
}

string query(string names[], int codes[], int ID) {
        for(short i = 0; i < 5; i++) {
                if(codes[i] == ID) {
                        return names[i];
                }
        }

        return "No hay registro";
}

int main() {
        int ID, codes[5];
        string names[5];

        input(names, codes);

        cout << "Ingrese el ID del empleado a consultar: ";
        cin >> ID;

        cout << query(names, codes, ID) << "\n";

        return 0;
}

File: GTP2.5/gtp1.py
def get_name(ID:int, ids:list[int], nombres:list[str]):
        for i in range(len(ids)):
                if ID == ids[i]:
                        return nombres[i]

        # if ID in ids:
        #         return nombres[ids.index(ID)]
        # else:
        #         print("No hay un empleado con ese ID")

def main():
        nombres:list[str] = []
        ids:list[int] = []

        for i in range(5):
                nombres.append(input(f"Ingrese el nombre del empleado {i}: "))
                ids.append(int(input("Ingrese el ID de este empleado: ")))
                print()
        
        name:str = get_name(input("Ingrese ID del empleado a consultar: "))

        if name:
                print(f"El nombre solicitado es {name}")



if __name__ == '__main__':
        main()
